app/interface.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'app'
+class Interface
+  def initialize
+    @app = App.new
+    puts '----Welcome to School Library App!----'
+    display_menu
+  end
+
+  def display_menu
+    loop do
+      print_menu
+      @option = gets.chomp.to_i
+      break if @option == 7
+
+      handle_option(@option)
+    end
+    handle_exit_option if @option == 7
+  end
+
+  def print_menu
+    menu_options = "
+1 - List all books
+2 - List all people
+3 - Create a person
+4 - Create a book
+5 - Create a rental
+6 - List all rentals for a given person id
+7 - Exit
+"
+    puts '', 'Please choose an option by entering a number:', menu_options
+  end
+
+  def handle_option(option)
+    case option
+    when 1
+      @app.list_all_books
+    when 2
+      @app.list_all_people
+    when 3
+      @app.create_person
+    when 4
```

Comment:
- [ ] When creating a person with `teacher` specilaization, i seem to be running into an error based on class parameter calls.
- Details of the log on this error can be found in the screenshot below 

### Error log:
![image](https://github.com/ahmidou09/OOP-school-library/assets/50990925/e4a6661c-d0ea-4b6c-95c2-47ff334a3cf2)